hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/HRCPrecompileSuite.java

```diff
@@ -0,0 +1,615 @@
+                hrcTokenAssociateFromSameEOATwiceShouldFail(),
+                hrcTokenDissociateWhenNotAssociatedShouldFail(),
+                hrcTokenDissociateWhenBalanceNotZeroShouldFail(),
+                hrcTooManyTokenAssociateShouldFail());
+    }
+
+    private HapiSpec hrcNftAndFungibleTokenAssociateFromEOA() {
+        final AtomicReference<String> fungibleTokenNum = new AtomicReference<>();
+        final AtomicReference<String> nonfungibleTokenNum = new AtomicReference<>();
+
+        return defaultHapiSpec("hrcNftAndFungibleTokenAssociateFromEOA")
+                .given(
+                        newKeyNamed(MULTI_KEY),
+                        newKeyNamed(RANDOM_KEY),
+                        cryptoCreate(ACCOUNT).balance(100 * ONE_HUNDRED_HBARS),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(FUNGIBLE_TOKEN)
+                                .tokenType(TokenType.FUNGIBLE_COMMON)
+                                .supplyType(TokenSupplyType.INFINITE)
+                                .initialSupply(5)
+                                .name(TOKEN_NAME)
+                                .treasury(TOKEN_TREASURY)
+                                .adminKey(MULTI_KEY)
+                                .supplyKey(MULTI_KEY)
+                                .exposingCreatedIdTo(fungibleTokenNum::set),
+                        tokenCreate(NON_FUNGIBLE_TOKEN)
+                                .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                                .name(TOKEN_NAME)
+                                .symbol(TOKEN_SYMBOL)
+                                .initialSupply(0)
+                                .treasury(TOKEN_TREASURY)
+                                .adminKey(MULTI_KEY)
+                                .supplyKey(MULTI_KEY)
+                                .exposingCreatedIdTo(nonfungibleTokenNum::set),
+                        uploadInitCode(HRC),
+                        contractCreate(HRC))
+                .when(withOpContext((spec, opLog) -> {
+                    var fungibleTokenAddress = asHexedSolidityAddress(asToken(fungibleTokenNum.get()));
+                    var nonfungibleTokenAddress = asHexedSolidityAddress(asToken(nonfungibleTokenNum.get()));
+                    allRunFor(
+                            spec,
+                            // Associate fungible token
+                            contractCallWithFunctionAbi(
+                                            fungibleTokenAddress,
+                                            getABIFor(
+                                                    com.hedera.services.bdd.suites.contract.Utils.FunctionType.FUNCTION,
+                                                    ASSOCIATE,
+                                                    HRC))
+                                    .payingWith(ACCOUNT)
+                                    .signingWith(RANDOM_KEY) // should still work when signed by random key
```

Comment:
This `signingWith()` modifier is a bit mysterious; but one can verify that it only changes the behavior of an `EthereumTransaction`. 😵‍💫 

So I'd suggest deleting it from these tests.